packages/components/src/overlays/Popover/src/Popover.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+    composeClassnameRenderProps,
+    SlotProvider,
+    cssModule,
+    ButtonContext,
+    ButtonGroupContext,
+    FooterContext,
+    TextContext,
+    HeadingContext,
+    LinkContext,
+    useColorSchemeContext, HopperProvider
+} from "@hopper-ui/components";
+import { type StyledComponentProps, useStyledSystem } from "@hopper-ui/styled-system";
+import clsx from "clsx";
+import { forwardRef, type ForwardedRef } from "react";
+import {
+    useContextProps,
+    type PopoverProps as RACPopoverProps,
+    type DialogTriggerProps,
+    DialogTrigger,
+    Popover as RACPopover,
+    composeRenderProps,
+    Dialog
+} from "react-aria-components";
+
+import { PopoverContext } from "./PopoverContext.ts";
+
+import styles from "./Popover.module.css";
+
+export const GlobalPopoverCssSelector = "hop-Popover";
+
+export interface PopoverProps extends StyledComponentProps<RACPopoverProps> {
+}
+
+export interface PopoverTriggerProps extends DialogTriggerProps {
+}
+
+export const PopoverTrigger = (props: PopoverTriggerProps) =>
+    <DialogTrigger {...props}>{props.children}</DialogTrigger>;
+
+function Popover(props: PopoverProps, ref: ForwardedRef<HTMLElement>) {
+    [props, ref] = useContextProps(props, ref, PopoverContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        children,
+        className,
+        style: styleProp,
+        offset = 4,
+        ...otherProps
+    } = ownProps;
+
+    const { colorScheme } = useColorSchemeContext();
+    
+    const popoverClassNames = composeClassnameRenderProps(
+        className,
+        GlobalPopoverCssSelector,
+        cssModule(
+            styles,
+            "hop-Popover"
+        )
+    );
+
+    const style = composeRenderProps(styleProp, prev => {
```

Comment:
If you are not adding anything to the style, you can simply let the style props be forwarded through ...otherprops. 

```diff
const {
        children,
        className,
-       style: styleProp,
        offset = 4,
        ...otherProps
    } = ownProps;
```

```diff
- const style = composeRenderProps(styleProp, prev => {
-     return {
-        ...prev
-     };
- });
```

```diff
<RACPopover
            {...otherProps}
            offset={offset}
            ref={ref}
            className={popoverClassNames}
-           style={style}
        >
```